### PR TITLE
Use AccessTraits instead of hardcoded View in DBSCAN

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: test/tstDBSCAN.cpp [hunk 1]
```diff
@@ -114,6 +122,27 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(dbscan, DeviceType, ARBORX_DEVICE_TYPES)
   using ArborX::Point;
   using ArborX::Details::verifyDBSCAN;
 
+  {
+    // Disable this test for CUDA, as it results in warning about calling
+    // __host__ function from __host__ __device__ one.
+#ifndef KOKKOS_ENABLE_CUDA
```
Comment (dalg24) [inline test/tstDBSCAN.cpp:128]: Use a view and relax this

**Review by dalg24 — CHANGES_REQUESTED** [review]

File: test/tstDBSCAN.cpp (hunk 1 above)
Comment (aprokop) [reply test/tstDBSCAN.cpp:128]: Just to make sure: are you proposing to use view, but make it a separate class, and specialize `AccessTraits` on it? The point here is to not use `View` for primitives.

Comment (dalg24) [reply test/tstDBSCAN.cpp:128]: Put it in a class if you like

Comment (aprokop) [reply test/tstDBSCAN.cpp]: Done.

**Review by masterleinad — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Comment by aprokop** [comment]
Usual two tests timeouts in SYCL (QueryTest and DistributedTest). Merging.

**Pull request merged**